lib/galaxy/tools/__init__.py

```diff
@@ -3228,7 +3229,7 @@ def add_copied_value_to_new_elements(new_tags_dict, dce):
                     tag_handler.add_tags_from_list(user=history.user, item=copied_value, new_tags_list=new_tags, flush=False)
             else:
                 # We have a collection, and we copy the elements so that we don't manipulate the original tags
-                copied_value = dce.element_object.copy(element_destination=history)
+                copied_value = dce.element_object.copy(copy_tags=dce.element_object.tags, element_destination=history)
```

Comment:
Same here, `DatasetCollection` instances don't have tag associations.
```suggestion
                copied_value = dce.element_object.copy(element_destination=history)
```